### PR TITLE
feat(parser): parse string interpolation

### PR DESCRIPTION
**Pull request by jlapacik**
### Done checklist
- [ ] docs/SPEC.md updated
- [x] Test cases written

### CODE REVIEW

File: ast/ast.go [hunk 1]
```diff
@@ -532,6 +535,56 @@ func (*StringLiteral) expression()          {}
...
+		ne.Parts = make([]*StringExpressionPart, len(e.Parts))
+		for i, p := range e.Parts {
+			ne.Parts[i] = p.Copy().(*StringExpressionPart)
+		}
+	}
+
+	return ne
+}
+
+type StringExpressionPart struct {
```
Comment (nathanielc) [inline ast/ast.go]: Should `StringExpressionPart` be an interface instead of a struct? It seems to behave like a union type.

Comment (jlapacik) [reply ast/ast.go]: I think a struct representing a string part makes more sense vs separate `Text` and `${ ... }` ast nodes. In my opinion.

**Comment by jlapacik** [comment]
@nathanielc yes I agree we should update the Rust parser too. I remembered the rust parts after I put up the PR. I'll have those updates soon.

**Comment by jlapacik** [comment]
Ported over string interpolation to the Rust parser as well.

File: ast/ast.go (hunk 1 above)
Comment (jlapacik) [reply ast/ast.go]: To keep parity with the Rust implementation where `StringExpressionPart` is implemented as an `enum` type, I did indeed have to make it an interface type in the Go implementation.

**Pull request merged**